docs/v1/api.md

```diff
@@ -0,0 +1,333 @@
+- `ServiceInstanceBinding`
+  - Created in response to a successfully executed `ServiceInstanceClaim`
+  - References credentials created as a result of the bind operation
+  - Holds a record of an application that created a `ServiceInstance` which
+    was successfully executed
+
+
+The below diagram shows the interactions between these resources:
+
+![Resource Interactions](./img/partial-flow.png)
+
+# Example Resources
+
+## `ServiceClass`
+
+```yaml
+kind: ServiceClass
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: postgres-small
+spec:
+  serviceID: c5a0ad8e-b57b…
+  planID: c68cd6b1-be9f…
+```
+
+## `ServiceInstance`
+
+```yaml
+kind: ServiceInstance
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: zip-code-db
+spec:
+  serviceClass: postgres-small
+```
+
+## Single Tenant / Dedicated Resources
+
+The `ServiceClassClaim` and `ServiceClassBinding` resources are created
+to give an application single-tenant, dedicated access to a backing resource.
+
+The developer should create a `ServiceClassClaim` resource, and the system
+should do the `provision` operation, then `bind` immediately thereafter.
+
+### ServiceClassClaim
+
+```yaml
+kind: ServiceClassClaim
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-app-db-claim
+  namespace: my-app
+status: unknown
+spec:
+  serviceClass: postgres-large
+  targetBinding: my-app-binding
+  status: unknown # only written by the service controller
+  statusDescription: unknown # only written by the service controller
+```
+
+### `ServiceClassBinding`
+
+This example shows the output of a single-tenant `ServiceInstanceClaim`.
+
+```yaml
+kind: ServiceInstanceBinding
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-app-binding # same as the previous 'targetBinding' field
+  namespace: my-app
+status:
+  serviceClass: postgres-large
+  secrets:
+    - my-zip-code-creds-password
+  configMaps:
+    - my-zip-code-creds-username
+    - my-zip-code-creds-conn-info
+```
+
+## Multi Tenant / Shared Resources
+
+The `ServiceInstanceClaim` and `ServiceInstanceBinding` resources are created
+to give an application multi-tenant, shared access to a backing resource.
+
+The developer should create a `ServiceInstanceClaim` resource, and the system
+should do the `bind` operation. It's assumed that an operator will have already
+created the `ServiceInstance` to which the `ServiceInstanceClaim` and
+`ServiceInstanceBinding` refers.
+
+
+### ServiceInstanceClaim
+
+```yaml
+kind: ServiceInstanceClaim
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: my-zip-code-db-claim
+  namespace: my-app
+spec:
+  serviceInstance: zip-code-db
+  targetBinding: my-zip-code-binding
+  status: unknown # only written by the service controller
+  statusDescription: unknown # only written by the service controller
+```
+
+### ServiceInstanceBinding
+
+This example shows the output of a multi-tenant `ServiceInstanceClaim`.
+
+```yaml
+kind: ServiceInstanceBinding
```

Comment:
Is the intent of ServiceInstanceBinding to record consumer state or producer state? 

Consumer state, such as the service name, secret(s), and configmap(s), could be written back to the claim, and those resources could be linked back to the claim, as well.

Are there other use cases for persisting the binding resources?